modules/executive/components/VoteModal/DefaultView.tsx

```diff
@@ -0,0 +1,350 @@
+import { useBreakpointIndex } from '@theme-ui/match-media';
+import BigNumber from 'bignumber.js';
+import { fetchJson } from 'lib/fetchJson';
+import getMaker, { getNetwork, personalSign } from 'lib/maker';
+import { sortBytesArray } from 'lib/utils';
+import { ANALYTICS_PAGES } from 'modules/app/client/analytics/analytics.constants';
+import { useAnalytics } from 'modules/app/client/analytics/useAnalytics';
+import SkeletonThemed from 'modules/app/components/SkeletonThemed';
+import useAccountsStore from 'modules/app/stores/accounts';
+import useTransactionsStore, { transactionsApi } from 'modules/app/stores/transactions';
+import { useExecutiveComments } from 'modules/comments/hooks/useExecutiveComments';
+import { ExecutiveCommentsRequestBody } from 'modules/comments/types/executiveComment';
+import { useAllSlates } from 'modules/executive/hooks/useAllSlates';
+import { useHat } from 'modules/executive/hooks/useHat';
+import { useMkrOnHat } from 'modules/executive/hooks/useMkrOnHat';
+import { useSpellData } from 'modules/executive/hooks/useSpellData';
+import { useVotedProposals } from 'modules/executive/hooks/useVotedProposals';
+import { CMSProposal, Proposal } from 'modules/executive/types';
+import { useLockedMkr } from 'modules/mkr/hooks/useLockedMkr';
+import React, { useEffect, useState } from 'react';
+import { Grid, Button, Flex, Close, Text, Box, Label, Checkbox, Textarea } from 'theme-ui';
+import shallow from 'zustand/shallow';
+
+export default function DefaultVoteModalView({
+  proposal,
+  onTransactionPending,
+  onTransactionMined,
+  onTransactionCreated,
+  onTransactionFailed
+}: {
+  proposal: Proposal;
+  onTransactionPending: () => void;
+  onTransactionMined: () => void;
+  onTransactionFailed: () => void;
+  onTransactionCreated: (txId: string) => void;
+}): React.ReactElement {
+  const { trackButtonClick } = useAnalytics(ANALYTICS_PAGES.EXECUTIVE);
+  const bpi = useBreakpointIndex();
+
+  const account = useAccountsStore(state => state.currentAccount);
+  const [voteProxy, voteDelegate] = useAccountsStore(state =>
+    account ? [state.proxies[account.address], state.voteDelegate] : [null, null]
+  );
+  const addressLockedMKR =
+    voteDelegate?.getVoteDelegateAddress() || voteProxy?.getProxyAddress() || account?.address;
+  const { data: lockedMkr, mutate: mutateLockedMkr } = useLockedMkr(
+    addressLockedMKR,
+    voteProxy,
+    voteDelegate
+  );
+  const { data: spellData, mutate: mutateSpellData } = useSpellData(proposal.address);
+
+  // revalidate on mount
+  useEffect(() => {
+    mutateLockedMkr();
+    mutateSpellData();
+  }, []);
+
+  const [hatChecked, setHatChecked] = useState(true);
+  const { data: currentSlate, mutate: mutateVotedProposals } = useVotedProposals();
```

Comment:
brought the currentSlate from the hook instead of nesting it 3 levels from the previous parent props